apps/support/benchmark.h

```diff
@@ -0,0 +1,26 @@
+#ifndef BENCHMARK_H
+#define BENCHMARK_H
+
+#include <chrono>
+#include <limits>
+
+// Benchmark the operation 'op'. The number of iterations refers to
+// how many times the operation is run for each time measurement, the
+// result is the minimum over a number of samples runs. The result is the
+// amount of time in seconds for one iteration.
+template <typename F>
+double benchmark(int samples, int iterations, F op) {
+    double best = std::numeric_limits<double>::infinity();
+    for (int i = 0; i < samples; i++) {
+        auto t1 = std::chrono::system_clock::now();
```

Comment:
I used system_clock because it seems to have the same accuracy as gettimeofday which we were using before, but we should probably just use high_resolution_clock anyways, so I changed it to that. high_resolution_clock is supposed to use the clock with the smallest ticks available on the system.